test/e2e/e2e.go

```diff
@@ -0,0 +1,164 @@
+)
+
+var (
+	operatorDeployed bool
+)
+
+func init() {
+	operatorDeployed = true
+}
+
+func RunE2ETests(t *testing.T) {
+	RegisterFailHandler(Fail)
+
+	RunSpecsWithDefaultAndCustomReporters(t,
+		"Controller Suite",
+		[]Reporter{printer.NewlineReporter{}})
+}
+
+var _ = Describe("validation", func() {
+	Context("general", func() {
+		It("should have the metal LB operator namespace", func() {
+			_, err := testclient.Client.Namespaces().Get(context.Background(), MetallbNameSpace, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("should have all the nodes in ready state", func() {
+			nodes, err := testclient.Client.Nodes().List(context.Background(), metav1.ListOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			for _, node := range nodes.Items {
+				nodeReady := false
+				for _, condition := range node.Status.Conditions {
+					if condition.Type == corev1.NodeReady &&
+						condition.Status == corev1.ConditionTrue {
+						nodeReady = true
+					}
+				}
+				Expect(nodeReady).To(BeTrue(), "Node ", node.Name, " is not ready")
+			}
+		})
+	})
+
+	Context("metallb", func() {
+		var expectedReadyReplicas int32
+
+		It("should have the metal LB operator deployment in running state", func() {
+			deploy, err := testclient.Client.Deployments(MetallbNameSpace).Get(context.Background(), MetallbOperatorDeploymentName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(deploy.Status.Replicas).To(Equal(deploy.Status.ReadyReplicas))
+
+			pods, err := testclient.Client.Pods(MetallbNameSpace).List(context.Background(), metav1.ListOptions{
+				LabelSelector: fmt.Sprintf("control-plane=%s", MetallbOperatorDeploymentLabel)})
+
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(pods.Items)).To(Equal(1))
+			Expect(pods.Items[0].Status.Phase).To(Equal(corev1.PodRunning))
+		})
+
+		It("should have the metal LB operator CRD available in the cluster", func() {
+			crd := &apiext.CustomResourceDefinition{}
+			err := testclient.Client.Get(context.Background(), goclient.ObjectKey{Name: MetallbOperatorCRDName}, crd)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("should have the metal LB operator pods in running state", func() {
+			metallb := &metallbv1alpha.Metallb{}
+			err := loadMetallbFromFile(metallb, "metallb.yaml")
+			Expect(err).ToNot(HaveOccurred())
+
+			By("installing metal LB operator", func() {
+				err = testclient.Client.Get(context.Background(), goclient.ObjectKey{Namespace: metallb.Namespace, Name: metallb.Name}, metallb)
+				if errors.IsNotFound(err) {
+					Expect(testclient.Client.Create(context.Background(), metallb)).Should(Succeed())
+					operatorDeployed = false
+				}
+			})
+
+			By("checking the metal LB operator controller deployment in running state", func() {
+				expectedReadyReplicas = 1
+				Eventually(func() int32 {
+					deploy, err := testclient.Client.Deployments(metallb.Namespace).Get(context.Background(), "controller", metav1.GetOptions{})
```

Comment:
should we set the deployment name as a const instead of using "controller" here?

Reply:
+1